mp/src/game/client/hl2/hud_credits.cpp

```diff
@@ -113,33 +70,50 @@ class CHudCredits : public CHudElement, public Panel
     bool m_bLastOneInPlace;
     int m_Alpha;
 
-    int m_iCreditsType;
-    int m_iLogoState;
+    float m_flCreditsPixelHeight;
+
+    int m_iMusicGUID;
+};
+
+using namespace vgui;
+
+static CreditsDialog *g_pCreditsDialog = nullptr;
 
-    float m_flFadeInTime;
-    float m_flFadeHoldTime;
-    float m_flFadeOutTime;
-    float m_flNextStartTime;
-    float m_flPauseBetweenWaves;
+//-----------------------------------------------------------------------------
+// Purpose: Constructor
+//-----------------------------------------------------------------------------
+CreditsDialog::CreditsDialog() : BaseClass(nullptr, "CreditsDialog")
+{
+    g_pCreditsDialog = this;
 
-    float m_flLogoTimeMod;
-    float m_flLogoTime;
-    float m_flLogoDesiredLength;
+    SetParent(enginevgui->GetPanel(PANEL_GAMEUIDLL));
+    SetProportional(true);
 
-    float m_flX;
-    float m_flY;
+    const auto settingsScheme = scheme()->LoadSchemeFromFile("resource/CreditsScheme.res", "CreditsScheme");
+    SetScheme(settingsScheme ? settingsScheme : scheme()->GetDefaultScheme());
```

Comment:
Panels without a scheme default to the default scheme, so this can just be an if:
```suggestion
    const auto settingsScheme = scheme()->LoadSchemeFromFile("resource/CreditsScheme.res", "CreditsScheme");
    if (settingsScheme)
    {
        SetScheme(settingsScheme);
    }
```

Reply:
I copied this from the settings dialog, didn't even rename it, woops. I did the requested change and changed that variable name to `creditsScheme`.